A3A/addons/patcom/functions/Patcom/fn_patrolGroupGarrison.sqf

```
@@ -29,53 +29,62 @@ params ["_group", "_position", "_radius"];
 private _units = units _group;
 private _buildings = [];
 private _newGroups = [];
+private _minimumUnits = 2; // Minimum units per building.
 
 if (count _units == 0) exitwith {};
 
 _group lockWP true;
-_buildings = nearestObjects [_position, keys PATCOM_Garrison_Positions, _radius];
+_buildings = [_position, _radius] call A3A_fnc_patrolEnterableBuildings;
 
-if (count _buildings == 0) then {
-    _buildings = [_position, _radius] call A3A_fnc_patrolEnterableBuildings;
-};
-
-// don't place units on destroyed buildings
+// Don't place units in destroyed buildings
 _buildings = _buildings select { damage _x < 1 && !isObjectHidden _x };
 _buildings = _buildings call BIS_fnc_arrayShuffle;
 
+// Figure out how many units should be put in each building.
+private _unitsPerBuilding = ceil(count _units / count _buildings);
+if (_unitsPerBuilding < _minimumUnits) then {_unitsPerBuilding = _minimumUnits};
+
 {
     if (count _units == 0) exitWith {};
     private _building = _x;
     private _class = typeOf _building;
     private _buildingPositions = [];
 
-    if (_class in PATCOM_Garrison_Positions) then {
+    // Check to see if building is in whitelist first for better unit positions.
+    if (_class in PATCOM_Garrison_Positions_Whitelist) then {
         {
             private _buildingPos = _building buildingPos _x;
             if !(_buildingPos isEqualTo [0,0,0]) then {
                 _buildingPositions pushBack _buildingPos;
             };
```

Should throw an error otherwise. It'll only happen if a whitelist entry is busted.

---

Resolved in [6ac8eda](https://github.com/official-antistasi-community/A3-Antistasi/pull/2876/commits/6ac8edad90e0aa28f073e0d8b6d957b8ef5a155b)